qiskit_research/utils/gates.py

```diff
@@ -136,6 +136,41 @@ def __array__(self, dtype=None):
         return np.array([[0, -1j], [1j, 0]], dtype=dtype)
 
 
+class PiPhiGate(Gate):
+    r"""The single-qubit X gate (:math:`\sigma_x`), implemented
+    with an angle :math:`\\phi` relative to the X-axis.
+    """
+
+    def __init__(self, phi: ParameterValueType, label: Optional[str] = None):
+        """Create new PiPhi gate."""
+        super().__init__("\\pi_{\\phi}", 1, [phi], label=label)
+        self.phi = phi
+
+    def _define(self):
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q, name=self.name)
+        # rules = [(U3Gate(pi, pi / 2, pi / 2), [q[0]], [])] # FIX
+        rules = [
+            (RZGate(self.phi), [q[0], []]),
+            (XGate(), [q[0]], []),
+            (RZGate(-self.phi), [q[0]], []),
+        ]  # maybe this is right
+        for instr, qargs, cargs in rules:
+            qc.append(instr, qargs, cargs)
+
+        self.definition = qc
+
+    def inverse(self):
+        r"""Return inverted Ym gate (:math:`Y{\dagger} = Y`)"""
+        return PiPhiGate(-self.phi)  # self-inverse
+
+    def __array__(self, dtype=None):
+        """Gate matrix."""
+        return np.cos(self.phi) * np.array([[0, 1], [1, 0]], dtype=dtype) + np.sin(
+            self.phi
+        ) * np.array([[0, -1j], [1j, 0]], dtype=dtype)
```

Comment:
Please add a test for this gate. Among other things, it should be tested that the matrix matches the gate decomposition.

Reply:
Added `test_piphi_definition` that compares the matrix (in `__array__`) and `definition`.